libs/core/src/components/pds-tooltip/pds-tooltip.tsx

```diff
@@ -133,54 +146,220 @@ export class PdsTooltip {
+   * If `htmlContent` is false, it attempts to find the actual element slotted as the trigger.
+   * Otherwise, or if no specific element is found, it defaults to the span wrapper around the trigger slot.
+   * This helps with precise alignment.
+   */
+  private determinePositioningAnchor(): HTMLElement | null {
+    let positioningAnchor: HTMLElement | null = this.triggerEl;
+
+    if (this.triggerEl !== null) {
+      const children = this.triggerEl.childNodes;
+      for (let i = 0; i < children.length; i++) {
+        const childNode = children[i];
+        if (childNode.nodeType === Node.ELEMENT_NODE) {
+          positioningAnchor = childNode as HTMLElement;
+          break; // Found the first element, use it as the anchor
+        }
+      }
+    }
+    // If no ELEMENT_NODE is found within this.triggerEl (e.g., if trigger is just text),
+    // positioningAnchor will correctly remain this.triggerEl (the span).
+    return positioningAnchor;
+  }
+
+  /**
+   * Centralized method to calculate and apply the tooltip's position.
+   * Uses the determined anchor element and the current content dimensions.
+   */
+  private repositionPortal() {
+    const anchor = this.determinePositioningAnchor();
+
+    if (anchor !== null && this.contentDiv !== null) {
+      positionTooltip({ elem: anchor, elemPlacement: this.placement, overlay: this.contentDiv });
+      const placementParts = this.placement.split('-');
+      const primaryPlacement = placementParts[0];
+      const isCardinalCenterPlacement = placementParts.length === 1;
+      if (isCardinalCenterPlacement) {
+        const anchorRect = anchor.getBoundingClientRect();
+        const overlayRect = this.contentDiv.getBoundingClientRect();
+        if (primaryPlacement === 'left' || primaryPlacement === 'right') {
+          const currentOverlayTop = parseFloat(this.contentDiv.style.top || '0');
+          const anchorCenterY = anchorRect.top + (anchorRect.height / 2);
+          const overlayCenterY = overlayRect.top + (overlayRect.height / 2);
+          const adjustmentY = anchorCenterY - overlayCenterY;
+          if (Math.abs(adjustmentY) > 0.5) {
+            this.contentDiv.style.top = `${currentOverlayTop + adjustmentY}px`;
+          }
+        } else if (primaryPlacement === 'top' || primaryPlacement === 'bottom') {
+          const currentOverlayLeft = parseFloat(this.contentDiv.style.left || '0');
+          const anchorCenterX = anchorRect.left + (anchorRect.width / 2);
+          const overlayCenterX = overlayRect.left + (overlayRect.width / 2);
+          const adjustmentX = anchorCenterX - overlayCenterX;
+          if (Math.abs(adjustmentX) > 0.5) {
+            this.contentDiv.style.left = `${currentOverlayLeft + adjustmentX}px`;
+          }
+        }
+      }
+    }
+  }
+
+  private createPortal() {
+    if (this.portalEl !== null) return;
+    this.portalEl = document.createElement('div');
```

Comment:
Add line break before and/or after the `if`'s

Reply:
Thanks for the help and clarification on these.